File: CASHIQ-CASHMANAGEMENT-MICROSERVICES/CASHIQ-CASHMANAGEMENT-MANAGE-CORE/src/main/java/com/cashiq/cashmanagement/entity/Transaction.java
package com.cashiq.cashmanagement.entity;

import com.cashiq.cashmanagement.enums.TransactionType;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "transactions")
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String description;

    private Double amount;

    private String category;

    private LocalDate date;

    private String paymentSource;

    @Enumerated(EnumType.STRING)
    private TransactionType type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private Users user;
}

File: CASHIQ-CASHMANAGEMENT-REACT-FRONTEND/src/models/Transaction.ts
export interface TransactionDTO {
    id?: number;
    description: string;
    amount: number;
    category: string;
    date: string; // YYYY-MM-DD
    paymentSource: string;
    type: 'INCOME' | 'EXPENSE' | 'DEPOSIT' | 'WITHDRAW';
}
